### PR TITLE
Angular 8 and Angular CLI build infrastructure

### PR DESCRIPTION
**Pull request by avatsaev**
### Changes

This MR introduces the correct way to build modern Angular Libraries by using the Angular CLI build infrastructure.
It also ensures the compatibility with future Angular versions as well as Angular Ivy compatibility.

- `projects/angular-jwt/` folder contains the library coude source
- `src` folder contains a mock app for the the library 
- unit testing happens in the mock app (`src/app/services/example-http.service.spec.ts`)

Running tests: `npm run ng test`

There aren't any major changes in the library source code, or any breaking changes.

The compiled library was tested on a large internal project, all unit tests passed, the application compiled to ivy without any errors, and authentication was working correctly wihtout any changes to the host application source code.

### References

- support ticket: https://github.com/auth0/angular2-jwt/issues/620

### Checklist

- [x] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
- [x] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)
- [x] All code quality tools/guidelines in the [CONTRIBUTING documentation](../CONTRIBUTING.md) have been run/followed
- [x] All relevant assets have been compiled as directed in the [CONTRIBUTING documentation](../CONTRIBUTING.md), if applicable

### CODE REVIEW

**Comment by avatsaev** [comment]
Will do!

**Comment by avatsaev** [comment]
Done, sorry about the readme file, missed it while copying, the instruction are the same, there is nothing to change.

**Comment by avatsaev** [comment]
@stevehobbsdev if you have any questions please let me know, will be happy to help

**Comment by stevehobbsdev** [comment]
Will do @avatsaev. It needs a discussion internally about how we go about this - thanks for your patience.

**Comment by Neetesh1** [comment]
@avatsaev could you please tell, when you are merging this PR?

**Comment by avatsaev** [comment]
@neetesh1 once the maintainers approve it, it's not up to me

**Comment by aecz** [comment]
@avatsaev Could you review the conflicts ?  Also I think you should  mark the conversation as resolved where @stevehobbsdev requested you made changes. After that, this PR will have all checks green and be ready to merge. Thanks!

**Comment by avatsaev** [comment]
@aecz I can't mark anything as resolved here, I just pushed the requested changes and asked for review again. Still waiting

**Comment by aecz** [comment]
@avatsaev Can you solve the branch conflicts at least ?

**Comment by avatsaev** [comment]
resolved.

**Comment by s992** [comment]
@stevehobbsdev Is there anything we can do to help get this across the line? This package is currently blocking our upgrade to Ivy, so would love to assist in any way we can.

**Comment by avatsaev** [comment]
Hey @s992 we couldn't wait for this to get reviewed soon enough so we deployed this version of the PR on npm in the meanwhile, been using it for a month or so, works very well, it's under the name of https://www.npmjs.com/package/@avatsaev/angular-jwt

I know it's not the best solution, but we had to move forward, if you're not comfortable using the npm package you can clone my repo, checkout the `angular-cli-refactor` branch, build and deploy the package on your private npm reg while we wait for this to be merged.

pinging @stevehobbsdev

**Comment by s992** [comment]
Thanks @avatsaev !

**Comment by stevehobbsdev** [comment]
Thanks for raising this again @avatsaev, and I do apologise for the mishap here.

We can bring your changes in and release as a new major version. However, to do that I will need you to undo your changes that switch out the `@auth0` namespace and remove the Auth0 info on the readme, plus anything else of that nature that is present in the PR.

Let me know when that is done 👍

**Comment by avatsaev** [comment]
@stevehobbsdev done, documents restored, I don't think i've missed anything

**Comment by mohyeid** [comment]
This is great, but there is 49 files changed. Can you summarize what changed? Or any reference what do we need to change to make our lib work? We are facing the same issue

**Comment by stevehobbsdev** [comment]
@avatsaev is best placed to provide that summary.

**Comment by avatsaev** [comment]
The lib sources themselves haven't changed, it's now just wrapped around Angular CLI builder. So the host apps don't need to change anything, just build the lib and ship the v4.

**Comment by stevehobbsdev** [comment]
This has now been published as v4, thanks again @avatsaev and apologies for the delay in getting this out.

**Comment by ltomes** [comment]
Hi, is any additional configuration required to use 4.0.0?
I'm running into the issues described on https://github.com/auth0/angular2-jwt/issues/605#issuecomment-587132443 with version 4.0.0 when trying to use the JwtHelperService (Before 4.0.0 I can't get Ivy to compile anyway).

Thanks

**Comment by stevehobbsdev** [comment]
@ltomes There should be no need for any additional configuration, no. With version 4, you should just be able to do `import { JwtHelperService } from '@auth0/angular-jwt'` and have it work. You would have no need for the workaround described in that thread (and like you say, the `src` directory is no longer included in the build).

If you have any specific error messages and/or a small project that reproduces the problem, I'd be happy to take a look.

**Comment by ltomes** [comment]
Hi Steve, thanks for confirming no other changes are required.
My issue appears unrelated to the `angular-jwt` update. My application is calling a different npm module (`openapi-generator-cli` generated code) after the helper service that fails to properly AOT compile (But gives no errors/warnings at compile time) in ivy mode.

The joys of updating multiple dependencies at once!
